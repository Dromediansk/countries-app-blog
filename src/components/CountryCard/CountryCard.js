import React from "react";
import "./CountryCard.css";

const CountryCard = ({ country }) => (
  <div className="country-card" data-testid={`${country.name}-card`}>
    <h3>{country.name}</h3>
    <hr />
    <ul className="country-info-list">
      <li className="country-info-item">
        <h4>Capital:</h4>
        <p>{country.capital}</p>
      </li>
      <li className="country-info-item">
        <h4>Population:</h4>
        <p>{country.population}</p>
      </li>
      <li className="country-info-item">
        <h4>Region:</h4>
        <p>{country.region}</p>
      </li>
      <li className="country-info-item">
        <img className="country-flag" src={country.flag} alt="Country flag" />
      </li>
    </ul>
  </div>
);

export default CountryCard;
