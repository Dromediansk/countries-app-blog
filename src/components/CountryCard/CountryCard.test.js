import React from "react";
import CountryCard from "./CountryCard";

import { render } from "@testing-library/react";

const stubbedCountry = {
  name: "Slovakia",
  population: 5000000,
  capital: "Bratislava",
  region: "Europe",
  flag: "Slovakia flag",
};

it("should match snapshot", () => {
  const { asFragment } = render(<CountryCard country={stubbedCountry} />);

  expect(asFragment()).toMatchSnapshot();
});
