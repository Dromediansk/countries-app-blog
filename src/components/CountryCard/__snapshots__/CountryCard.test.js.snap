// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="country-card"
    data-testid="Slovakia-card"
  >
    <h3>
      Slovakia
    </h3>
    <hr />
    <ul
      class="country-info-list"
    >
      <li
        class="country-info-item"
      >
        <h4>
          Capital:
        </h4>
        <p>
          Bratislava
        </p>
      </li>
      <li
        class="country-info-item"
      >
        <h4>
          Population:
        </h4>
        <p>
          5000000
        </p>
      </li>
      <li
        class="country-info-item"
      >
        <h4>
          Region:
        </h4>
        <p>
          Europe
        </p>
      </li>
      <li
        class="country-info-item"
      >
        <img
          alt="Country flag"
          class="country-flag"
          src="Slovakia flag"
        />
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
